### PR TITLE
Add feedback processing

### PR DESCRIPTION
**Pull request by LukeVincentCivica**
- Update the notifications services to send notifications for the feedback

- store whether the feedback option was selected in the database

### CODE REVIEW

File: src/services/notifications.service.js [hunk 1]
```diff
@@ -176,7 +157,9 @@ const sendEmailOfType = async (
     );
     throw err;
   }
-  return emailSent;
+  statusEmitter.emit("incrementCount", "emailNotificationsSucceeded");
+  statusEmitter.emit("setStatus", "mostRecentEmailNotificationSucceeded", true);
+  logEmitter.emit("functionSuccess", "registration.service", "sendEmailOfType");
 };
 
 /**
```
Comment (michalrak) [inline src/services/notifications.service.js]: JsDoc not updated

File: src/services/notifications.service.js [hunk 2]
```diff
@@ -191,35 +174,52 @@ const sendNotifications = async (
...
 
-  await sendEmailOfType(
-    "FBO",
+  emailsToSend.push({
+    type: "FBO",
+    address: fboEmailAddress,
+    templateId: configData.notify_template_keys.fbo_submission_complete
+  });
+
+  if (registration.metadata.feedback1) {
```
Comment (michalrak) [inline src/services/notifications.service.js]: There is no unit test for this path

File: src/services/notifications.service.js [hunk 3]
```diff
@@ -108,59 +108,40 @@ const transformDataForNotify = (
  * @returns {object} Object that returns email sent status and recipients email address
  */
 
-const sendEmailOfType = async (
-  typeOfEmail,
+const sendEmails = async (
```
[the remark this answers is not in the document]
Comment (LukeVincentCivica) [reply src/services/notifications.service.js]: JsDoc updated

File: src/services/notifications.service.js (hunk 1 above)
Comment (LukeVincentCivica) [reply src/services/notifications.service.js]: JsDoc now updated

File: src/services/notifications.service.js (hunk 2 above)
Comment (LukeVincentCivica) [reply src/services/notifications.service.js]: Unit test wasn't running due to incorrect beforeEach set up, those tests now running

File: src/services/notifications.service.js [hunk 4]
```diff
@@ -108,59 +108,40 @@ const transformDataForNotify = (
...
-    );
-
-    let pdfFile = undefined;
-    if (typeOfEmail === "LC") {
-      pdfFile = await pdfGenerator(dataForPDF);
+    for (let email in emailsToSend) {
+      let pdfFile = undefined;
+
+      if (emailsToSend[email].type === "LC") {
+        const dataForPDF = transformDataForPdf(
```
[the remark this answers is not in the document]
Comment (LukeVincentCivica) [reply src/services/notifications.service.js]: Updated where the PDF get's created and now only assigns the document in the for loop

File: src/services/notifications.service.js [hunk 5]
```diff
@@ -176,50 +156,69 @@ const sendEmailOfType = async (
     );
     throw err;
   }
-  return emailSent;
+  statusEmitter.emit("incrementCount", "emailNotificationsSucceeded");
+  statusEmitter.emit("setStatus", "mostRecentEmailNotificationSucceeded", true);
+  logEmitter.emit("functionSuccess", "registration.service", "sendEmailOfType");
```
Comment (michalrak) [inline src/services/notifications.service.js]: This function is not called "sendEmailOfType" anymore, so log calls need to be updated.

File: src/services/notifications.service.js [hunk 6]
```diff
@@ -176,50 +156,69 @@ const sendEmailOfType = async (
     );
     throw err;
   }
-  return emailSent;
+  statusEmitter.emit("incrementCount", "emailNotificationsSucceeded");
```
Comment (michalrak) [inline src/services/notifications.service.js]: Each of these statements could just be moved back inside try block for consistency

File: src/services/notifications.service.js (hunk 1 above)
Comment (michalrak) [reply src/services/notifications.service.js]: Function description still needs to be updated

File: src/services/notifications.service.js (hunk 6 above)
Comment (LukeVincentCivica) [reply src/services/notifications.service.js]: done

**Pull request merged**